rest/client.go

```diff
@@ -74,10 +79,11 @@ func NewClient(httpClient Doer, options ...func(*Client)) *Client {
 	}
 
 	c := &Client{
-		httpClient:    httpClient,
-		Endpoint:      endpoint,
-		RateLimitFunc: defaultRateLimitFunc,
-		UserAgent:     defaultUserAgent,
+		httpClient:       httpClient,
```

Comment:
If you use `httpClient: http.DefaultClient` instead, you can drop the `httpClient Doer` argument and the default invocation of `NewClient` just becomes: `rest.NewClient()`

Reply:
This is definitely better, but a breaking change that doesn't seem worth it at the moment.